### PR TITLE
Navigation library

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1204374977096192/f

### Description
Implement the navigation library, see [asana task](https://app.asana.com/0/488551667048375/1204374977096192/f) for more info

### Steps to test this PR
NA, nav library not used yet

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3055** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3055" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3055?utm_source=stack-comment).

File: navigation/navigation-api/src/main/java/com/duckduckgo/navigation/api/GlobalActivityStarter.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+     * ```kotlin
+     * @ContributeToActivityStarter(ExampleActivityParams::class)
+     * class ExampleActivity() : DuckDuckGoActivity() {...}
+     * ```
+     */
+    interface ParamToActivityMapper {
+        /**
+         * @return the [Activity] class if it matches the desired parameters, otherwise return `null`
+         */
+        fun map(activityParams: ActivityParams): Class<out AppCompatActivity>?
```
Comment (malmstein) [inline navigation/navigation-api/src/main/java/com/duckduckgo/navigation/api/GlobalActivityStarter.kt:99]: should this be `DuckDuckGoActivity` instead? We would avoid adding app-compat as dependency to the module.

File: navigation/navigation-impl/build.gradle [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+dependencies {
+    anvil project(':anvil-compiler')
+
+    implementation project(':navigation-api')
+    implementation project(':di')
+
+    implementation Kotlin.stdlib.jdk7
+    implementation Google.dagger
+    implementation "com.squareup.logcat:logcat:_"
```
Comment (malmstein) [inline navigation/navigation-impl/build.gradle:33]: nit: is this the new `Timber`?

**Review by malmstein — APPROVED** [review]
Just a nit about app-compat, looks great though. Thanks for taking care of it.

**Pull request merged**